test/unit-test/FreeRTOS_Stream_Buffer/FreeRTOS_Stream_Buffer_utest.c

```diff
@@ -0,0 +1,1108 @@
+ *        zero offset.
+ */
+void test_uxStreamBufferAdd_BufferFullZeroOffset( void )
+{
+    /* Now we need to get a buffer which can store up to 1024 bytes. */
+    StreamBuffer_t * pxLocalBuffer = malloc( sizeof( StreamBuffer_t ) - sizeof( pxLocalBuffer->ucArray ) + 1024 );
+    size_t uxOffset = 0;
+    uint8_t pucData[ 512 ];
+    size_t uxByteCount = 0;
+    size_t uxReturn;
+
+    pxLocalBuffer->LENGTH = 1024;
+    pxLocalBuffer->uxHead = 100;
+    pxLocalBuffer->uxTail = pxLocalBuffer->uxHead + 1;
+    pxLocalBuffer->uxMid = 0;
+    pxLocalBuffer->uxFront = 0;
+
+    FreeRTOS_min_uint32_Stub( FreeRTOS_min_stub );
+
+    uxReturn = uxStreamBufferAdd( pxLocalBuffer, uxOffset, pucData, uxByteCount );
+
+    TEST_ASSERT_EQUAL( 0, uxReturn );
+}
+
+/*
+ * @brief Test adding to the buffer when it is already full with
+ *        positive offset.
+ */
+void test_uxStreamBufferAdd_BufferFullPositiveOffset( void )
+{
+    /* Now we need to get a buffer which can store up to 1024 bytes. */
+    StreamBuffer_t * pxLocalBuffer = malloc( sizeof( StreamBuffer_t ) - sizeof( pxLocalBuffer->ucArray ) + 1024 );
+    size_t uxOffset = 10;
+    uint8_t pucData[ 512 ];
+    size_t uxByteCount = 0;
+    size_t uxReturn;
+
+    pxLocalBuffer->LENGTH = 1024;
+    pxLocalBuffer->uxHead = 100;
+    pxLocalBuffer->uxTail = pxLocalBuffer->uxHead + 1;
+    pxLocalBuffer->uxMid = 0;
+    pxLocalBuffer->uxFront = 0;
+
+    FreeRTOS_min_uint32_Stub( FreeRTOS_min_stub );
+
+    uxReturn = uxStreamBufferAdd( pxLocalBuffer, uxOffset, pucData, uxByteCount );
+
+    TEST_ASSERT_EQUAL( 0, uxReturn );
+}
+
+/*
+ * @brief Test adding to the buffer when it has less space than data with
+ *        zero offset.
+ */
+void test_uxStreamBufferAdd_BufferHasLessSpaceThanData_ZeroOffset( void )
+{
+    /* Now we need to get a buffer which can store up to 1024 bytes. */
+    StreamBuffer_t * pxLocalBuffer = malloc( sizeof( StreamBuffer_t ) - sizeof( pxLocalBuffer->ucArray ) + 1024 );
+    size_t uxOffset = 0;
+    uint8_t pucData[ 512 ];
+    size_t uxByteCount = sizeof( pucData );
+    size_t uxReturn;
+
+    /* Clear everything. */
+    memset( pxLocalBuffer, 0, sizeof( StreamBuffer_t ) - sizeof( pxLocalBuffer->ucArray ) + 1024 );
+
+    /* Fill the Data (0xAB) */
+    memset( pucData, 0xAB, sizeof( pucData ) );
+
+    pxLocalBuffer->LENGTH = 1024;
+    /* Set Head at 0 for simplicity. */
+    pxLocalBuffer->uxHead = 0;
+    /* Only 500 bytes available. */
+    pxLocalBuffer->uxTail = pxLocalBuffer->uxHead + 500 + 1;
+    pxLocalBuffer->uxMid = 0;
+    pxLocalBuffer->uxFront = 0;
+
+    FreeRTOS_min_uint32_Stub( FreeRTOS_min_stub );
+
+    uxReturn = uxStreamBufferAdd( pxLocalBuffer, uxOffset, pucData, uxByteCount );
+
+    /* Only 500 bytes should be written. */
+    TEST_ASSERT_EQUAL( 500, uxReturn );
+
+    /* Assert that data was indeed written. */
+    TEST_ASSERT_EQUAL_MEMORY( pxLocalBuffer->ucArray + uxOffset, pucData, 500 - uxOffset );
```

Comment:
Should the last parameter be uxByteCount? Also assert that other bytes in the buffer remained unchanged.